accounts/abi/bind/base.go

```diff
@@ -123,7 +136,12 @@ func (c *BoundContract) Call(opts *CallOpts, result interface{}, method string,
 	if err != nil {
 		return err
 	}
-	output, err := c.caller.ContractCall(opts.Context, c.address, input, opts.Pending)
+	// Create the call message we use to make the call
+	callMsg := ethereum.CallMsg{
+		To:   c.address,
+		Data: input,
+	}
+	output, err := c.backend.CallContract(opts.Context, callMsg, nil)
```

Comment:
@karalabe I didn't really look at this previously, the caching here looks unsafe. The contract creation can get reverted at any time, in which case the check will not be executed. The call will still fail, but not with ErrNoCode.

IMHO it would be safer to do the check after the call.